src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt

```diff
@@ -230,23 +226,43 @@ internal class RecommendationService(
   }
 
   @OptIn(ExperimentalStdlibApi::class)
-  fun generatePartA(recommendationId: Long, username: String?, userEmail: String?): DocumentResponse {
-
+  suspend fun generatePartA(recommendationId: Long, username: String?, userEmail: String?): DocumentResponse {
     val recommendationEntity = updateRecommendation(null, recommendationId, username, userEmail, true)
     val userAccessResponse = recommendationEntity.data.crn?.let { userAccessValidator.checkUserAccess(it) }
     if (userAccessValidator.isUserExcludedOrRestricted(userAccessResponse)) {
       throw UserAccessException(Gson().toJson(userAccessResponse))
     } else {
-      val fileContents = templateReplacementService.generateDocFromRecommendation(recommendationEntity, DocumentType.PART_A_DOCUMENT)
+      val updatedRecommendation = updateRecommendation(recommendationEntity)
+      val fileContents = templateReplacementService.generateDocFromRecommendation(updatedRecommendation, DocumentType.PART_A_DOCUMENT)
       return DocumentResponse(
-        fileName = generateDocumentFileName(recommendationEntity.data, "NAT_Recall_Part_A"),
+        fileName = generateDocumentFileName(updatedRecommendation.data, "NAT_Recall_Part_A"),
         fileContents = fileContents
       )
     }
   }
 
-  private fun generateDocumentFileName(recommendation: RecommendationModel, prefix: String): String {
+  suspend fun updateRecommendation(recommendationEntity: RecommendationEntity): RecommendationEntity {
```

Comment:
We now have 2 methods called updateRecommendation in the same class. Is it worth renaming one so we know when we should use one over the other? Or adding a comment so we clear know the purpose of each one? We should probably move so they appear in the same place in the code as well so it is clear there are 2, what you think?

Reply:
I have renamed the one that just adds the extra fields for clarity.